### ISSUE TITLE
Problem with TestCommandLineInvocationSameAsConfig 

### ISSUE THREAD

**Issue by pnemade** [ansible/ansible-lint#309]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible --version
2.4.2.0
ansible-lint --version
3.4.19
```

Using rhel7 machine with latest packages from epel7

# Desired Behaviour
tests should run without a problem on RHEL machines as well

# Actual Behaviour (Bug report only)
```
======================================================================
FAIL: test_config_can_be_overridden (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 68, in test_config_can_be_overridden
    no_override = self.run_ansible_lint(args="-t bad_tag")
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 32, in run_ansible_lint
    self.assertFalse(err, "Expected no error but was " + str(err))
AssertionError: Expected no error but was Traceback (most recent call last):
  File "../bin/ansible-lint", line 4, in <module>
    import ansiblelint.__main__
ImportError: No module named ansiblelint.__main__
======================================================================
```

**Comment by pnemade** [comment]
Got a build failure https://koji.fedoraproject.org/koji/taskinfo?taskID=23636143
This issue is introduced by commit in PR #290

**Comment by pnemade** [comment]
I think I can fix this by changing
```
%check
%{__python2} setup.py test
```
to
```
%check
PYTHONPATH=%{buildroot}%{python2_sitelib}  %{__python2} setup.py test
```

but then I am not sure why without adding that worked fine for Fedora builds only and not for epel7 build.

**Comment by ari3s** [comment]
I can see other TestCommandLineInvocationSameAsConfig test failures of the same ansible and ansible-lint versions mentioned above when I try to build Python 3 package of ansible-lint in Fedora:

```
+ /usr/bin/python3 setup.py test
running test
running egg_info
writing lib/ansible_lint.egg-info/PKG-INFO
writing dependency_links to lib/ansible_lint.egg-info/dependency_links.txt
writing entry points to lib/ansible_lint.egg-info/entry_points.txt
writing requirements to lib/ansible_lint.egg-info/requires.txt
writing top-level names to lib/ansible_lint.egg-info/top_level.txt
reading manifest file 'lib/ansible_lint.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'lib/ansible_lint.egg-info/SOURCES.txt'
running build_ext
test_rule_matching (test.TestLintRule.TestRule) ... ok
test_rule_postmatching (test.TestLintRule.TestRule) ... ok
test_file_negative (test.TestEnvVarsInCommand.TestEnvVarsInCommand) ... ok
test_file_positive (test.TestEnvVarsInCommand.TestEnvVarsInCommand) ... ok
test_file_positive_2_4 (test.TestEnvVarsInCommand.TestEnvVarsInCommand) ... ok
test_file_negative (test.TestUseHandlerRatherThanWhenChanged.TestUseHandlerRatherThanWhenChanged) ... ok
test_file_positive (test.TestUseHandlerRatherThanWhenChanged.TestUseHandlerRatherThanWhenChanged) ... ok
test_dict_format_line (test.TestFormatter.TestFormatter) ... ok
test_format_coloured_string (test.TestFormatter.TestFormatter) ... ok
test_unicode_format_string (test.TestFormatter.TestFormatter) ... ok
test_block_included_tasks (test.TestTaskIncludes.TestTaskIncludes) ... ok
test_block_included_tasks_with_rescue_and_always (test.TestTaskIncludes.TestTaskIncludes) ... ok
test_import_tasks_2_4_style (test.TestTaskIncludes.TestTaskIncludes) ... ok
test_include_tasks_2_4_style (test.TestTaskIncludes.TestTaskIncludes) ... ok
test_include_tasks_with_block_include (test.TestTaskIncludes.TestTaskIncludes) ... ok
test_included_tasks (test.TestTaskIncludes.TestTaskIncludes) ... ok
test_role_handler_positive (test.TestRoleHandlers.TestRoleHandlers) ... ok
test_file_negative (test.TestOctalPermissions.TestOctalPermissionsRuleWithFile) ... ok
test_file_positive (test.TestOctalPermissions.TestOctalPermissionsRuleWithFile) ... ok
test_file_negative (test.TestAlwaysRunRule.TestAlwaysRun) ... ok
test_file_positive (test.TestAlwaysRunRule.TestAlwaysRun) ... ok
test_config_can_be_overridden (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig) ... FAIL
test_different_config_file (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig) ... FAIL
test_exclude (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig) ... FAIL
test_parseable_as_config (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig) ... FAIL
test_quiet_as_config (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig) ... FAIL
test_rulesdir_as_config (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig) ... FAIL
test_skip_list (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig) ... FAIL
test_tags (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig) ... FAIL
test_use_default_rules (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig) ... FAIL
test_verbosity (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig) ... FAIL
test_runner_count (test.TestSkippedTasks.TestRule) ... ok
test_package_not_latest_negative (test.TestPackageIsNotLatest.TestPackageIsNotLatestRule) ... ok
test_package_not_latest_positive (test.TestPackageIsNotLatest.TestPackageIsNotLatestRule) ... ok
test_dir_with_fullpath (test.TestRunner.TestRule) ... ok
test_dir_with_trailing_slash (test.TestRunner.TestRule) ... ok
test_files_not_scanned_twice (test.TestRunner.TestRule) ... ok
test_runner_become_count (test.TestRunner.TestRule) ... ok
test_runner_block_count (test.TestRunner.TestRule) ... WARNING: Couldn't open /builddir/build/BUILD/ansible-lint-3.4.19/test/doesnotexist.yml - No such file or directory
ok
test_runner_count (test.TestRunner.TestRule) ... ok
test_runner_empty_tags_count (test.TestRunner.TestRule) ... ok
test_runner_excludes_paths (test.TestRunner.TestRule) ... ok
test_unicode_parseable_colored_formatting (test.TestRunner.TestRule) ... ok
test_unicode_quiet_colored_formatting (test.TestRunner.TestRule) ... ok
test_unicode_runner_count (test.TestRunner.TestRule) ... ok
test_unicode_standard_color_formatting (test.TestRunner.TestRule) ... ok
test_unicode_standard_formatting (test.TestRunner.TestRule) ... ok
test_command_changes_negative (test.TestCommandHasChangesCheck.TestCommandHasChangesCheck) ... ok
test_command_changes_positive (test.TestCommandHasChangesCheck.TestCommandHasChangesCheck) ... ok
test_file_negative (test.TestBecomeUserWithoutBecome.TestBecomeUserWithoutBecome) ... ok
test_file_positive (test.TestBecomeUserWithoutBecome.TestBecomeUserWithoutBecome) ... ok
test_file_negative (test.TestUseCommandInsteadOfShell.TestUseCommandInsteadOfShell) ... ok
test_file_positive (test.TestUseCommandInsteadOfShell.TestUseCommandInsteadOfShell) ... ok
test_file_negative (test.TestUsingBareVariablesIsDeprecated.TestUsingBareVariablesIsDeprecated) ... ok
test_file_positive (test.TestUsingBareVariablesIsDeprecated.TestUsingBareVariablesIsDeprecated) ... ok
test_existing_filter_on_unknown_var (test.TestUtils.TestUtils) ... ok
test_extract_from_list (test.TestUtils.TestUtils) ... ok
test_missing_filter (test.TestUtils.TestUtils) ... ok
test_normalize_args (test.TestUtils.TestUtils) ... ok
test_normalize_complex_command (test.TestUtils.TestUtils) ... ok
test_normalize_simple_command (test.TestUtils.TestUtils) ... ok
test_simple_template (test.TestUtils.TestUtils) ... ok
test_task_to_str_unicode (test.TestUtils.TestUtils) ... ok
test_tokenize_blank (test.TestUtils.TestUtils) ... ok
test_tokenize_command_with_args (test.TestUtils.TestUtils) ... ok
test_tokenize_more_than_one_arg (test.TestUtils.TestUtils) ... ok
test_tokenize_single_word (test.TestUtils.TestUtils) ... ok
test_tokenize_string_module_and_arg (test.TestUtils.TestUtils) ... ok
test_tokenize_strips_action (test.TestUtils.TestUtils) ... ok
test_check_value_nested (test.TestSudoRule.TestSudoRule) ... /builddir/build/BUILD/ansible-lint-3.4.19/test/TestSudoRule.py:44: DeprecationWarning: Please use assertEqual instead.
  self.assertEquals(2, len(result))
ok
test_check_value_shallow_dict (test.TestSudoRule.TestSudoRule) ... ok
test_check_value_simple_matching (test.TestSudoRule.TestSudoRule) ... ok
test_matchplay_sudo (test.TestSudoRule.TestSudoRuleWithFile) ... /builddir/build/BUILD/ansible-lint-3.4.19/test/TestSudoRule.py:54: ResourceWarning: unclosed file <_io.TextIOWrapper name='test/sudo.yml' mode='r' encoding='UTF-8'>
  yaml = ansiblelint.utils.parse_yaml_linenumbers(open(self.file1).read(), self.file1)
ok
test_matchtask_sudo (test.TestSudoRule.TestSudoRuleWithFile) ... /builddir/build/BUILD/ansible-lint-3.4.19/test/TestSudoRule.py:62: ResourceWarning: unclosed file <_io.TextIOWrapper name='test/sudo.yml' mode='r' encoding='UTF-8'>
  yaml = ansiblelint.utils.parse_yaml_linenumbers(open(self.file1).read(), self.file1)
ok
test_load_collection_from_directory (test.TestRulesCollection.TestRulesCollection) ... ok
test_no_duplicate_rule_ids (test.TestRulesCollection.TestRulesCollection) ... ok
test_run_collection (test.TestRulesCollection.TestRulesCollection) ... ok
test_skip_id (test.TestRulesCollection.TestRulesCollection) ... ok
test_skip_non_existent_id (test.TestRulesCollection.TestRulesCollection) ... ok
test_skip_tags (test.TestRulesCollection.TestRulesCollection) ... ok
test_tags (test.TestRulesCollection.TestRulesCollection) ... ok

======================================================================
FAIL: test_config_can_be_overridden (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 68, in test_config_can_be_overridden
    no_override = self.run_ansible_lint(args="-t bad_tag")
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 32, in run_ansible_lint
    self.assertFalse(err, "Expected no error but was " + str(err))
AssertionError: b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n' is not false : Expected no error but was b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n'

======================================================================
FAIL: test_different_config_file (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 77, in test_different_config_file
    diff_config = self.run_ansible_lint(args="-c ./subdir/ansible-config.yml")
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 32, in run_ansible_lint
    self.assertFalse(err, "Expected no error but was " + str(err))
AssertionError: b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n' is not false : Expected no error but was b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n'

======================================================================
FAIL: test_exclude (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 65, in test_exclude
    self.assert_config_for("--exclude ../test/", dict(exclude_paths=["../test/"]))
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 37, in assert_config_for
    with_arg = self.run_ansible_lint(args=cli_arg)
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 32, in run_ansible_lint
    self.assertFalse(err, "Expected no error but was " + str(err))
AssertionError: b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n' is not false : Expected no error but was b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n'

======================================================================
FAIL: test_parseable_as_config (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 43, in test_parseable_as_config
    self.assert_config_for("-p", dict(parseable=True))
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 37, in assert_config_for
    with_arg = self.run_ansible_lint(args=cli_arg)
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 32, in run_ansible_lint
    self.assertFalse(err, "Expected no error but was " + str(err))
AssertionError: b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n' is not false : Expected no error but was b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n'

======================================================================
FAIL: test_quiet_as_config (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 46, in test_quiet_as_config
    self.assert_config_for("-q", dict(quiet=True))
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 37, in assert_config_for
    with_arg = self.run_ansible_lint(args=cli_arg)
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 32, in run_ansible_lint
    self.assertFalse(err, "Expected no error but was " + str(err))
AssertionError: b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n' is not false : Expected no error but was b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n'

======================================================================
FAIL: test_rulesdir_as_config (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 49, in test_rulesdir_as_config
    self.assert_config_for("-r ../test/rules/", dict(rulesdir=["../test/rules/"]))
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 37, in assert_config_for
    with_arg = self.run_ansible_lint(args=cli_arg)
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 32, in run_ansible_lint
    self.assertFalse(err, "Expected no error but was " + str(err))
AssertionError: b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n' is not false : Expected no error but was b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n'

======================================================================
FAIL: test_skip_list (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 62, in test_skip_list
    self.assert_config_for("-x bad_tag", dict(skip_list=["bad_tag"]))
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 37, in assert_config_for
    with_arg = self.run_ansible_lint(args=cli_arg)
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 32, in run_ansible_lint
    self.assertFalse(err, "Expected no error but was " + str(err))
AssertionError: b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n' is not false : Expected no error but was b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n'

======================================================================
FAIL: test_tags (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 56, in test_tags
    self.assert_config_for("-t skip_ansible_lint", dict(tags=["skip_ansible_lint"]))
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 37, in assert_config_for
    with_arg = self.run_ansible_lint(args=cli_arg)
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 32, in run_ansible_lint
    self.assertFalse(err, "Expected no error but was " + str(err))
AssertionError: b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n' is not false : Expected no error but was b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n'

======================================================================
FAIL: test_use_default_rules (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 53, in test_use_default_rules
    use_default_rules=True))
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 37, in assert_config_for
    with_arg = self.run_ansible_lint(args=cli_arg)
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 32, in run_ansible_lint
    self.assertFalse(err, "Expected no error but was " + str(err))
AssertionError: b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n' is not false : Expected no error but was b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n'

======================================================================
FAIL: test_verbosity (test.TestCommandLineInvocationSameAsConfig.TestCommandLineInvocationSameAsConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 59, in test_verbosity
    self.assert_config_for("-v", dict(verbosity=1))
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 37, in assert_config_for
    with_arg = self.run_ansible_lint(args=cli_arg)
  File "/builddir/build/BUILD/ansible-lint-3.4.19/test/TestCommandLineInvocationSameAsConfig.py", line 32, in run_ansible_lint
    self.assertFalse(err, "Expected no error but was " + str(err))
AssertionError: b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n' is not false : Expected no error but was b'Traceback (most recent call last):\n  File "../bin/ansible-lint", line 4, in <module>\n    import ansiblelint.__main__\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/__init__.py", line 28, in <module>\n    import ansiblelint.utils\n  File "/builddir/build/BUILD/ansible-lint-3.4.19/lib/ansiblelint/utils.py", line 26, in <module>\n    from ansible import constants\n  File "/usr/lib/python3.6/site-packages/ansible/constants.py", line 18, in <module>\n    from ansible.config.manager import ConfigManager, ensure_type, get_ini_config_value\n  File "/usr/lib/python3.6/site-packages/ansible/config/manager.py", line 11, in <module>\n    import yaml\n  File "/usr/lib64/python3.6/site-packages/yaml/__init__.py", line 284\n    class YAMLObject(metaclass=YAMLObjectMetaclass):\n                              ^\nSyntaxError: invalid syntax\n'

----------------------------------------------------------------------
Ran 81 tests in 5.276s

FAILED (failures=10)
Test failed: <unittest.runner.TextTestResult run=81 errors=0 failures=10>
error: Test failed: <unittest.runner.TextTestResult run=81 errors=0 failures=10>
error: Bad exit status from /var/tmp/rpm-tmp.zVCqmv (%check)
    Bad exit status from /var/tmp/rpm-tmp.zVCqmv (%check)


RPM build errors:
ERROR: Exception(/home/makerpm/rpmbuild/SRPMS/ansible-lint-3.4.19-2.fc26.src.rpm) Config(fedora-rawhide-x86_64) 1 minutes 36 seconds
INFO: Results and/or logs in: /var/lib/mock/fedora-rawhide-x86_64/result
ERROR: Command failed: 
 # /usr/bin/systemd-nspawn -q -M d8ac41be1a28474e91d28d2cb84eaabf -D /var/lib/mock/fedora-rawhide-x86_64/root -a --setenv=TERM=vt100 --setenv=SHELL=/bin/bash --setenv=HOME=/builddir --setenv=HOSTNAME=mock --setenv=PATH=/usr/bin:/bin:/usr/sbin:/sbin --setenv=PROMPT_COMMAND=printf "\033]0;<mock-chroot>\007" --setenv=PS1=<mock-chroot> \s-\v\$  --setenv=LANG=en_US.utf8 -u mockbuild bash --login -c /usr/bin/rpmbuild -bb --target x86_64 --nodeps /builddir/build/SPECS/ansible-lint.spec
```

**Comment by willthames** [comment]
@ari3s - your errors appear to from `import yaml` (that error looks like python2 is trying to load a python3 yaml)

If either of you have any suggestions for fixing these or improving my tests to catch these, let me know.

**Comment by pnemade** [comment]
Currently I am busy with some other work but if I get some free time, I will have a look at this issue.
@ari3s I will for now remove this test from execution and will build new release.

**Comment by pnemade** [comment]
I actually looked more into the test execution and based on above comment from @willthames I got some fix which can now make that test execution successful for both py2 and py3 environment. I am just now adjusting the proper environment in bin/ansible-lint by replacing existing shebang with proper python versioned shebang
```
sed -i 's|/usr/bin/env python|%{_bindir}/python2|' bin/ansible-lint
PYTHONPATH=%{buildroot}%{python2_sitelib} %{__python2} setup.py test
sed -i 's|%{_bindir}/python2|%{_bindir}/python3|' bin/ansible-lint
PYTHONPATH=%{buildroot}%{python3_sitelib} %{__python3} setup.py test
```

**Comment by willthames** [comment]
Thanks @pnemade - if there's a better way I should be doing this, let me
know - my understanding is `/usr/bin/env python` is the most portable
mechanism for now

**Comment by webknjaz** [comment]
I believe this doesn't affect the current tests suite so I'm closing this. But FTR removing `bin/ansible-lint` is tracked @ #748.
